tasks/componentUpdateTasks.ts

```diff
@@ -0,0 +1,186 @@
+    if (printCommand) {
+        console.log(stdout);
+    }
+    return Promise.resolve(stdout);
+}
+
+gulp.task('publish roslyn copilot', async () => {
+    const parsedArgs = minimist<Options>(process.argv.slice(2));
+
+    // Get staging directory from environment variable passed from pipeline
+    const stagingDir = process.env['STAGING_DIRECTORY'];
+    if (!stagingDir) {
+        console.log('STAGING_DIRECTORY environment variable not set; skipping package.json update.');
+        return;
+    }
+
+    if (!fs.existsSync(stagingDir)) {
+        console.log(`Staging directory not found at ${stagingDir}; skipping package.json update.`);
+        return;
+    }
+
+    // Find the Roslyn zip file in the staging directory (we know it was copied here)
+    const files = fs.readdirSync(stagingDir);
+    const zipFile = files.find((file) => /Roslyn\.LanguageServer.*\.zip$/i.test(file));
+
+    if (!zipFile) {
+        console.log(`No Roslyn LanguageServer zip file found in ${stagingDir}; skipping package.json update.`);
+        return;
+    }
+
+    const zipPath = path.join(stagingDir, zipFile);
+    console.log(`Using zip file: ${zipPath}`);
+    const zipName = zipFile;
+
+    // Extract version from file name
+    // Example: "Microsoft.VisualStudio.Copilot.Roslyn.LanguageServer-18.0.671-alpha.zip"
+    let version: string | null = null;
+    const m = zipName.match(/Microsoft\.VisualStudio\.Copilot\.Roslyn\.LanguageServer-(.+)\.zip$/i);
+    if (m && m[1]) {
+        version = m[1];
+    }
+
+    if (!version) {
+        console.log(`Could not extract version from file name ${zipName}; skipping.`);
+        return;
+    }
+
+    console.log(`Extracted version: ${version}`);
+
+    const pkg = getPackageJSON();
+    let updated = false;
+
+    if (pkg.runtimeDependencies && Array.isArray(pkg.runtimeDependencies)) {
+        for (let i = 0; i < pkg.runtimeDependencies.length; i++) {
+            const dep = pkg.runtimeDependencies[i];
+            if (dep && dep.id === 'RoslynCopilot') {
+                const oldUrl = dep.url as string;
+                const newUrl = oldUrl.replace(
+                    /Microsoft\.VisualStudio\.Copilot\.Roslyn\.LanguageServer-[^/]+?\.zip/,
+                    `Microsoft.VisualStudio.Copilot.Roslyn.LanguageServer-${version}.zip`
+                );
+                if (oldUrl !== newUrl) {
+                    pkg.runtimeDependencies[i].url = newUrl;
+                    updated = true;
+                    console.log(`Updated RoslynCopilot url:\n  ${oldUrl}\n-> ${newUrl}`);
+                } else {
+                    console.log('RoslynCopilot url already up to date.');
+                }
+                break;
+            }
+        }
+    }
+
+    if (!updated) {
+        console.log('No changes required to package.json; aborting PR creation.');
+        return;
+    }
+
+    // Persist package.json
+    fs.writeFileSync('package.json', JSON.stringify(pkg, null, 2) + '\n', { encoding: 'utf8' });
```

Comment:
Does this preserve ordering of properties and objects?

Reply:
Switched to using an existing function for this